journal/src/test/java/io/camunda/zeebe/journal/file/SegmentedJournalReaderTest.java

```diff
@@ -108,6 +108,35 @@ void shouldSeekToAnyAsqnInMultipleSegments() {
     }
   }
 
+  @Test
+  void shouldReaderCannotReadWhenAccessingDeletedSegment() {
+    // given
+    journal.append(data);
+    final var reader = journal.openReader();
+
+    // when
+    journal.reset(100);
+
+    // then
+    assertThat(reader.hasNext()).isFalse();
+  }
+
+  @Test
+  void shouldReaderCanReadAfterReset() {
```

Comment:
:upside_down_face: `shouldReadAfterReset`? This method name looks funny now, but it's understandable, so consider this a nit pick :smile:

Reply:
Thanks for the name suggestion. I was finding it strange too. But I had a hard time coming up with something else. :relieved: :sweat_smile: